src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateForResult = { ...state };
+
+  for (const action in actions) {
+    if (actions[action].type === 'addProperties') {
+      const addValues = actions[action].extraData;
+
+      for (const addValue in addValues) {
+        stateForResult[addValue] = addValues[addValue];
+      }
+      result.push(stateForResult);
+    } else if (actions[action].type === 'removeProperties') {
+      const removeKeys = actions[action].keysToRemove;
+
+      for (const removeKey of removeKeys) {
+        delete stateForResult[removeKey];
+      }
+      result.push(stateForResult);
+    } else if (actions[action].type === 'clear') {
+      for (const stateKey in stateForResult) {
+        delete stateForResult[stateKey];
+      }
+      result.push(stateForResult);
+    }
```

Comment:
please get rid of `else if` blocks. just do:
```suggestion
  if (actions[action].type === 'addProperties') {
    const addValues = actions[action].extraData;

    for (const addValue in addValues) {
      stateForResult[addValue] = addValues[addValue];
    }
    result.push(stateForResult);
  }
  
  if (actions[action].type === 'removeProperties') {
    const removeKeys = actions[action].keysToRemove;

    for (const removeKey of removeKeys) {
      delete stateForResult[removeKey];
    }
    result.push(stateForResult);
  }
  
  if (actions[action].type === 'clear') {
    for (const stateKey in stateForResult) {
      delete stateForResult[stateKey];
    }
    result.push(stateForResult);
  }
```

Reply:
or better to use switch case. chose the one you like more